mirrord/layer/src/error.rs

```diff
@@ -142,6 +143,9 @@ pub(crate) enum LayerError {
 
     #[error("mirrord-layer: Got unexpected response error from agent: {0}")]
     UnexpectedResponseError(ResponseError),
+
+    #[error("mirrord-layer: Stolen HTTP request forwarding failed with `{0}`.")]
+    HttpForwardingError(#[from] HttpForwarderError),
```

Comment:
`HttpForwardingError` is a `LayerError`, that all `HttpForwarderError`s are converted to when converted to `LayerError`, this is just for having the the HTTP forwarding related errors grouped under their own error type instead of directly in the `LayerError`. @meowjesty says we should try not to have huge error enums, but break them down into an error hierarchy.